gcp/api/server.py

```diff
@@ -658,29 +762,54 @@ def query_by_version(context: grpc.ServicerContext,
   if ecosystem:
     query = query.filter(osv.Bug.ecosystem == ecosystem)
 
+  # Don't filter by ecosystem if it's a purl query
+  if purl and not ecosystem:
+    purl_ecosystem = purl_helpers.purl_to_ecosystem(purl.type)
+    if purl_ecosystem:
+      ecosystem = purl_ecosystem
+  else:
+    logging.warning('ecosystem specified in a purl query')
```

Comment:
I'm not sure if this ever makes sense. We can probably just return an error here.